### PR TITLE
Unified filter and table interface

### PR DESCRIPTION
**Pull request by nick-lehmann**
Combines filters for both scalar and vectorized operations into the same class hierarchy. This leads into a unified interface for the table that allows for both scalar and vectorized filtering, while the remaining code (especially for setup) can remain the same. All individual table implementations have been refactored, but not implementation specific helpers like the `IntermediateTable` of the `RowStore`.

This refactor should allow us for simpler testing and benchmarking code.

### CODE REVIEW

File: src/pax_store/main.cpp [hunk 1]
```diff
@@ -169,73 +169,50 @@ void testBasicAVXFilters() {
...
-//     __mmask16 match(__m512i reg, __mmask16 mask = ONE_MASK) override { return _mm512_mask_cmpeq_epi32_mask(mask, reg, value_register); };
-//     __mmask8 match(__m512i reg, __mmask8 mask = ONE_MASK) override { return _mm512_mask_cmpeq_epi64_mask(mask, reg, value_register); };
-// };
-
-template <typename T, bool AVX>
-class Filter {
-   public:
-    virtual bool match() = 0;
-};
+enum class SIMD { Scalar, AVX512 };
```
Comment (vincentadamthefirst) [inline src/pax_store/main.cpp]: Maybe rename `Scalar` to `None` since scalar is not really SIMD.

File: src/include/SIMD.h [hunk 2]
```diff
@@ -0,0 +1,3 @@
+#pragma once
+
+enum class SIMD { Scalar, AVX512 };
```
Comment (vincentadamthefirst) [inline src/include/SIMD.h:3]: Maybe rename `Scalar` to `None` since scalar execution is not SIMD.

Comment (nick-lehmann) [reply src/include/SIMD.h]: Done

**Comment by adbexy** [comment]
Amazing Work! I've just removed some references to deleted files from the CMakeLists. So in my opinion, it's ready to merge.

**Pull request merged**